### PR TITLE
Upgrade to Python3

### PR DESCRIPTION
**Pull request by rebkwok**
This upgrades for python 3.6 compatability.

Also drops support for python 2.6 and Django <1.8, and adds tox and travis.

### CODE REVIEW

File: django_hashedfilenamestorage/tests.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def patch(namespace, **values):
         if namespace._wrapped is None:
             namespace._setup()
         namespace = namespace._wrapped
-    for (name, value) in values.iteritems():
+    for (name, value) in list(values.items()):
```
Comment (hlwill) [inline django_hashedfilenamestorage/tests.py]: Does this need to be cast to a list?

Comment (rebkwok) [reply django_hashedfilenamestorage/tests.py]: No, that's 2to3 being over-cautious again; will fix

File: django_hashedfilenamestorage/tests.py [hunk 2]
```diff
@@ -110,16 +110,16 @@ def test_save(self):
             name1 = storage.save('foo/bar.txt', ContentFile(self.CONTENT))
             self.assertEqual(name1,
                              'foo/%s.txt' % self.SHA1SUM)
-            self.assertEqual(storage.open(name1).read(), self.CONTENT)
+            self.assertEqual(storage.open(name1, 'rt').read(), self.CONTENT)
```
Comment (pauricthelodger) [inline django_hashedfilenamestorage/tests.py]: Is `'rt'` not the default for the storage already in Python 3?

**Comment by pauricthelodger** [comment]
@rebkwok, can you close PRs #2 and #4 as well once this is merged? =)

File: django_hashedfilenamestorage/tests.py (hunk 2 above)
Comment (rebkwok) [reply django_hashedfilenamestorage/tests.py]: Yes, but the test fails under python 2 if it isn't specified

Comment (pauricthelodger) [reply django_hashedfilenamestorage/tests.py]: Ah, of course, cheers for reminder =)

**Comment by rebkwok** [comment]
@pauricthelodger Yes - also added another commit to fix issues #3 and #5

**Comment by rebkwok** [comment]
Fixes #3 
Fixes #5

**Pull request merged**